### PR TITLE
Simulation + other updates

### PR DESCRIPTION
**Pull request by virtuald**
Includes CANifier (no UI) and barebones TalonSRX/VictorSPX simulation support (with sim UI).

### CODE REVIEW

**Comment by auscompgeek** [comment]
\o/

File: gen/hooks.py [hunk 1]
```diff
@@ -99,6 +99,10 @@ def function_hook(fn, data):
     # Python exposed function name conveerted to camelcase
     x_name = m.group(1)
     x_name = x_name[0].lower() + x_name[1:]
+    
+    # Make create functions less visible
+    if x_name.startswith('create'):
+        x_name = '_' + x_name
```
Comment (auscompgeek) [inline gen/hooks.py]: There's actually a bunch of different functions that are internal, like the versioned motion profiling functions. I think this should live in the autogen YAML configs.

Comment (virtuald) [reply gen/hooks.py]: Maybe an 'internal: true' flag in the YAML then?

**Comment by virtuald** [comment]
Sadly, the return types aren't being handled properly when it's an array being returned, so this doesn't compile right now. Will try to fix tomorrow night.

**Comment by auscompgeek** [comment]
We should take this opportunity to mark the `*_2` methods as internal too.

**Comment by virtuald** [comment]
Done, anything else needed here? Otherwise I think I'm going to push a 2018.1.0 release.

**Comment by auscompgeek** [comment]
I think this looks sane.

**Pull request merged**